doc/replication-policies.md

```diff
@@ -0,0 +1,197 @@
+
+Note however, that in the following example we'll use the `startWithTaskGroup:error:` method of
+`CDTReplicator` instead of `startWithError:` so that we can batch the replication tasks and wait
+for the completion of all of them.
+
+For example:
+```objc
+/** Start both a push and pull replication.
+ *
+ *  @param taskGroup A dispatch_group_t to allow us to wait for replications to complete.
+ */
+- (void)startReplications:(dispatch_group_t)taskGroup
+{
+    // Create and start the replicator -- -start is essential!
+    CDTReplicatorFactory *replicatorFactory =
+    [[CDTReplicatorFactory alloc] initWithDatastoreManager:manager];
+
+    // username/password can be Cloudant API keys
+    NSString *s = @"https://username:password@username.cloudant.com/my_database";
+    NSURL *remoteDatabaseURL = [NSURL URLWithString:s];
+    CDTDatastore *datastore = [manager datastoreNamed:@"my_datastore"];
+
+    // Create a replicator that replicates changes from the local
+    // datastore to the remote database.
+    CDTPushReplication *pushReplication = [CDTPushReplication replicationWithSource:datastore
+                                                                             target:remoteDatabaseURL];
+
+    // Create a replicator that replicates changes from a remote
+    // database to the local one.
+    CDTPullReplication *pullReplication = [CDTPullReplication replicationWithSource:remoteDatabaseURL
+                                                                             target:datastore];
+
+    NSError *pushError;
+    CDTReplicator *pushReplicator = [replicatorFactory oneWay:pushReplication
+                                        sessionConfigDelegate:self
+                                                        error:&error];
+
+    NSError *pullError;
+    CDTReplicator *pullReplicator = [replicatorFactory oneWay:pullReplication
+                                        sessionConfigDelegate:self
+                                                        error:&error];
+
+    // Start the push replication
+    if (![pushReplicator startWithTaskGroup:taskGroup error:&error]){
+        //handle error
+    }
+
+    // Start the pull replication
+    if (![pullReplicator startWithTaskGroup:taskGroup error:&error]){
+        //handle error
+    }
+}
+```
+
+Create a method to start replications and wait for their completion (on a background thread).
+
+```objc
+-(void)syncInBackgroundWithCompletionHandler:(void (^)())completionHandler
+{
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
+        dispatch_group_t backgroundTasks = dispatch_group_create();
+
+        UIBackgroundTaskIdentifier taskId = [[UIApplication sharedApplication] beginBackgroundTaskWithExpirationHandler:NULL];
+
+        [self startReplications:backgroundTasks];
+
+        dispatch_group_wait(backgroundTasks, dispatch_time(DISPATCH_TIME_NOW, 30000000000));
```

Comment:
Maybe use NSEC_PER_SEC? The documentation for this stuff is amazingly evasive at telling us if it's measure in µs or something else, but that's what I'm guessing.